planning/freespace_planner/test/test_freespace_planner_node_interface.cpp

```diff
@@ -36,24 +44,50 @@ TEST(PlanningModuleInterfaceTest, testPlanningInterfaceWithVariousTrajectoryInpu
     {"--ros-args", "--params-file",
      planning_test_utils_dir + "/config/test_vehicle_info.param.yaml", "--params-file",
      freespace_planner_dir + "/config/freespace_planner.param.yaml"});
-  auto test_target_node = std::make_shared<freespace_planner::FreespacePlannerNode>(node_options);
+  return std::make_shared<FreespacePlannerNode>(node_options);
+}
 
+void publishMandatoryTopics(
+  std::shared_ptr<PlanningInterfaceTestManager> test_manager,
+  rclcpp::Node::SharedPtr test_target_node)
+{
   // publish necessary topics from test_manager
   test_manager->publishTF(test_target_node, "/tf");
   test_manager->publishOdometry(test_target_node, "freespace_planner/input/odometry");
   test_manager->publishOccupancyGrid(test_target_node, "freespace_planner/input/occupancy_grid");
   test_manager->publishParkingScenario(test_target_node, "freespace_planner/input/scenario");
+}
 
-  // set subscriber with topic name: freespace_planner → test_node_
-  test_manager->setRouteInputTopicName("freespace_planner/input/route");
+TEST(PlanningModuleInterfaceTest, testPlanningInterfaceWithVariousTrajectoryInput)
+{
+  rclcpp::init(0, nullptr);
 
-  // set freespace_planner's input topic name(this topic is changed to test node)
-  test_manager->setTrajectorySubscriber("freespace_planner/output/trajectory");
+  auto test_manager = generateTestManager();
+  auto test_target_node = generateNode();
+  publishMandatoryTopics(test_manager, test_target_node);
 
   // test with normal route
   ASSERT_NO_THROW(test_manager->testWithBehaviorNominalRoute(test_target_node));
   EXPECT_GE(test_manager->getReceivedTopicNum(), 1);
 
   // test with empty route
   ASSERT_NO_THROW(test_manager->testWithAbnormalRoute(test_target_node));
+  rclcpp::shutdown();
+}
+
+TEST(PlanningModuleInterfaceTest, NodeTestWithOffTrackEgoPose)
+{
+  rclcpp::init(0, nullptr);
+  auto test_manager = generateTestManager();
+  auto test_target_node = generateNode();
+
+  publishMandatoryTopics(test_manager, test_target_node);
+
+  // test for normal route
+  ASSERT_NO_THROW(test_manager->testWithBehaviorNominalRoute(test_target_node));
+  EXPECT_GE(test_manager->getReceivedTopicNum(), 1);
+
```

Comment:
@TakaHoribe 
```
test_manager->setInitialPoseTopicName("freespace_planner/input/odometry");
```
is necessary to define `input_initial_pose_name_`
And with this change, colcon test passed

Reply:
```
test_manager->setInitialPoseTopicName("freespace_planner/input/odometry");
```
is better to be in `generateTestManager()`

Reply:
Thank you. Addressed in https://github.com/autowarefoundation/autoware.universe/pull/3593/commits/1d029c461aa2acbea34f33ca240d7c71a8aa783d